File: core/src/main/java/de/qabel/core/accounting/BoxClient.java
package de.qabel.core.accounting;

import de.qabel.core.exceptions.QblCreateAccountFailException;
import de.qabel.core.exceptions.QblInvalidCredentials;
import org.apache.http.HttpRequest;
import org.apache.http.client.utils.URIBuilder;

import java.io.IOException;
import java.util.ArrayList;

public interface BoxClient extends RequestAuthorizer {


    void login() throws IOException, QblInvalidCredentials;

    QuotaState getQuotaState() throws IOException, QblInvalidCredentials;

    void updatePrefixes() throws IOException, QblInvalidCredentials;

    void createPrefix() throws IOException, QblInvalidCredentials;

    URIBuilder buildUri(String resource);

    URIBuilder buildBlockUri(String resource);

    ArrayList<String> getPrefixes() throws IOException, QblInvalidCredentials;

    AccountingProfile getProfile();

    void resetPassword(String email) throws IOException;

    void createBoxAccount(String email) throws IOException, QblCreateAccountFailException;
}

File: core/src/main/java/de/qabel/core/accounting/BoxClient.kt
package de.qabel.core.accounting

import de.qabel.core.exceptions.QblCreateAccountFailException
import de.qabel.core.exceptions.QblInvalidCredentials
import org.apache.http.HttpRequest
import org.apache.http.client.utils.URIBuilder

import java.io.IOException
import java.util.ArrayList

interface BoxClient {


    @Throws(IOException::class, QblInvalidCredentials::class)
    fun login()

    val quotaState: QuotaState

    @Throws(IOException::class, QblInvalidCredentials::class)
    fun authorize(request: HttpRequest)

    @Throws(IOException::class, QblInvalidCredentials::class)
    fun updatePrefixes()

    @Throws(IOException::class, QblInvalidCredentials::class)
    fun createPrefix()

    fun buildUri(resource: String): URIBuilder

    fun buildBlockUri(resource: String): URIBuilder

    val prefixes: ArrayList<String>

    val profile: AccountingProfile

    @Throws(IOException::class)
    fun resetPassword(email: String)

    @Throws(IOException::class, QblCreateAccountFailException::class)
    fun createBoxAccount(email: String)
}
